### PR TITLE
ISSUE-172 Map /root/.m2 to TMPFS

### PR DESCRIPTION
**Pull request by chibenwa**
This saves significant compilation/resource copy time...

### CODE REVIEW

**Review by trantienduchn — APPROVED** [review]

**Review by Arsnael — APPROVED** [review]

File: workflow-job [hunk 1]
```diff
@@ -573,7 +573,7 @@ def buildVolumeFromTag(def node) {
 def compile() {
     sh "docker build --tag=${images.jamesCompile} dockerfiles/compilation/java-8"
 
-    def tmpfsOptions = '--tmpfs=/tmp:rw,exec,size=1g --tmpfs=/james-project:rw,size=4g'
+    def tmpfsOptions = '--tmpfs=/tmp:rw,exec,size=1g --tmpfs=/root/.m2:rw,exec,size=1g --tmpfs=/james-project:rw,size=4g'
```
Comment (trantienduchn) [inline workflow-job:576]: what does `size=1g` mean? If it is the size limitation then I think you should increase, my .m2 directory takes above 2GB

Comment (chibenwa) [reply workflow-job:576]: But your .m2 directory have not the scope of a single build... You have many version of many different dependencies, on many different projects.

Comment (chibenwa) [reply workflow-job:576]: Total size is about 350 MO (local compilation), then 1 GO is faaar from enough

**Review by aduprat — APPROVED** [review]

**Comment by mbaechler** [comment]
any clue it's actually faster?

**Comment by chibenwa** [comment]
> any clue it's actually faster?

Will try

**Comment by chibenwa** [comment]
Just tried and compared with current version: we are CPU bound, m2 seems to be cached by the filesystem, gains are just a few seconds

**Pull request closed without merging**